src/protocols/secure_channel/CASESession.cpp

```diff
@@ -262,8 +227,9 @@ CASESession::ListenForSessionEstablishment(OperationalCredentialSet * operationa
 }
 
 CHIP_ERROR CASESession::EstablishSession(const Transport::PeerAddress peerAddress,
-                                         OperationalCredentialSet * operationalCredentialSet, NodeId peerNodeId, uint16_t myKeyId,
-                                         ExchangeContext * exchangeCtxt, SessionEstablishmentDelegate * delegate)
+                                         OperationalCredentialSet * operationalCredentialSet, uint8_t opCredSetIndex,
+                                         NodeId peerNodeId, uint16_t myKeyId, ExchangeContext * exchangeCtxt,
+                                         SessionEstablishmentDelegate * delegate)
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
 
```

Comment:
this method seems to mix Return and Exit.  Should the return on exchangeCtxt be an exit so that the code does 'Clear" ?

Reply:
No, because when returning on exchangeCtxt the class wouldn't have been initialized yet.